Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1792,6 +1793,7 @@ open class NavigationMapView: UIView {
                 // find the topmost non symbol layer for layerIdentifier in lowermostSymbolLayers.
                 if targetLayer == nil,
                    layerInfo.type.rawValue != "symbol",
+                   !layerInfo.id.contains("copy"),
```

Comment:
Some style has `copy` layer along with other layers at the same strata. It will lead to the wrong topmost non symbol layer.

Reply:
Style layer IDs can be arbitrarily named by the style designer; the presence of “copy” means nothing in particular, so we shouldn’t hard-code this magic value. Is there anything else we can look for to determine whether a non-symbol layer should be treated like a symbol layer for this purpose? If not, the application should specify a custom layer position.

Reply:
Thanks for the suggestion, the `copy` is replaced with a general `label` in case of this situation would happen. And also the custom layer position for the route line is supported during active navigation and when route line related properties change.

Reply:
The navigation SDK shouldn’t impose _any_ requirements on the layer ID. The layer ID is intended to be freeform text. If a team’s iOS developer is merely integrating a style that a designer authored in Mapbox Studio, we have no opportunity to communicate to the style designer that this exceptional layer should have `label` in the ID. On the other hand, if the iOS developer is adding the layer using runtime styling, they might still want to call it something that doesn’t mention “label”.

Do we know more about the use cases where a circle layer would imitate a symbol layer? Can we rely on `circlePitchAlignment` to be `viewport`?

Reply:
I removed the `id` check. But I don't think the circle layer is the only `copy` layer. The check around the `viewport` is not definition. When testing the test app, if the styles are not fully loaded, there may be random `copy` layers.

In Nav SDK, on both mobile and CarPlay, we have listeners for `MapView.mapboxMap.onNext(event: .styleLoaded, handler:)`, when the style changes, The route line layers are updated and regenerated.
If the style is not fully loaded, there may be some random `copy` layers, or the right layer position has been found but we couldn't add it. 

So the best way to solve this issue, is that the users provide their own `customLayerPosition` for the route line. This PR would remember the custom layer position for route line in both mobile and CarPlay. When the style changes, the route line would use the previous remembered custom layer position for the new style by default.

Reply:
I opened #4079 to track enhancing the default positioning heuristic to account for circle layers.

Reply:
I adopted #4079 to check the `circle` layer as a label layer. Route line will be placed below them as default.